src/witan/httpapi/components/activities.clj

```diff
@@ -409,27 +442,89 @@
          (create-error! db command-id file-id (-> payload :reason name)))
         (log/error "Could't find a metadatastore ID in payload:" payload)))))
 
+(defmethod on-event
+  [:kixi.datastore/files-added-to-bundle "1.0.0"]
+  [db event]
+  (when-let [command-id (:kixi.command/id event)]
+    (when-let [receipt (retrieve-receipt db command-id)]
+      (if-let [file-id (get-metadatastore-id event)]
+        (complete-receipt!
+         db
+         command-id
+         nil #_(str "/api/datapacks/" file-id "/metadata"))
+        (log/error "Could't find a metadatastore ID in payload:" event)))))
+
+(defmethod on-event
+  [:kixi.datastore/files-add-to-bundle-rejected "2.0.0"]
+  [db event]
+  (when-let [command-id (:kixi.command/id event)]
+    (when-let [receipt (retrieve-receipt db command-id)]
+      (if-let [file-id (get-metadatastore-id event)]
+        (complete-receipt!
+         db
+         command-id
+         (create-error! db command-id file-id (name (or (:reason event)
+                                                        (:kixi.event.bundle.addfiles.rejection/reason event)))))
+        (log/error "Could't find a metadatastore ID in payload:" event)))))
+
+(defmethod on-event
+  [:kixi.datastore/files-removed-from-bundle "1.0.0"]
+  [db event]
+  (when-let [command-id (:kixi.command/id event)]
+    (when-let [receipt (retrieve-receipt db command-id)]
+      (if-let [file-id (get-metadatastore-id event)]
```

Comment:
this error handling feels a bit repetitive. It would be something to revisit if we ever have to refactor this for new/different functionality.